src/main/scala/is/hail/io/reference/FASTAReader.scala

```diff
@@ -0,0 +1,185 @@
+package is.hail.io.reference
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+import java.util
+import java.util.Map.Entry
+
+import htsjdk.samtools.reference.{FastaSequenceIndex, FastaSequenceIndexCreator, ReferenceSequenceFile, ReferenceSequenceFileFactory}
+import is.hail.HailContext
+import is.hail.utils._
+import is.hail.variant.{RGBase, ReferenceGenome, Locus}
+import org.apache.hadoop
+
+import scala.language.postfixOps
+import scala.sys.process._
+
+class SerializableReferenceSequenceFile(val hConf: SerializableHadoopConfiguration, val fastaFile: String, val indexFile: String) extends Serializable {
+  @transient var value_ : ReferenceSequenceFile = _
+
+  def value: ReferenceSequenceFile = {
+    if (value_ == null) {
+      val localFastaFile = FASTAReader.getLocalFastaFileName(hConf, fastaFile, indexFile)
+      value_ = ReferenceSequenceFileFactory.getReferenceSequenceFile(new java.io.File(localFastaFile))
+    }
+    value_
+  }
```

Comment:
Is there a compelling reason not to use `lazy val` here? That's basically scala's built-in, thread-safe way to lazily initialize a value.

```scala
lazy val value = {
  val localFastaFile = FASTAReader.getLocalFastaFileName(hConf, fastaFile, indexFile)
  ReferenceSequenceFileFactory.getReferenceSequenceFile(new java.io.File(localFastaFile))
}
```

Reply:
I can't do this because the class won't be serializable.

Reply:
I'm pretty sure you can add `@transient` to a `lazy val`. Does that not work?